src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs

```diff
@@ -9,35 +9,33 @@ namespace Microsoft.VisualBasic.Tests
 {
     public class FinancialTests
     {
-        private static bool IsNotArmOrAlpine() => !PlatformDetection.IsArmOrArm64Process && !PlatformDetection.IsAlpine;
+        private static readonly int precision = PlatformDetection.IsFullFramework || PlatformDetection.IsArmOrArm64Process || PlatformDetection.IsAlpine ? 14 : 15;
 
         /// <summary>
-        /// The accuracy of some numeric parsing and formatting has been increased,
-        /// so this will use correct value if the tests run on Full Framework.
+        /// The accuracy of some numeric parsing and formatting depends on the platform.
         /// </summary>
-        private static void AreEqual<T>(T expectedOld, T expectedNew, T actual)
+        private static void AreEqual(double expected, double actual)
         {
-            T expected = PlatformDetection.IsFullFramework ? expectedOld : expectedNew;
-            Assert.Equal(expected, actual);
+            Assert.Equal(expected, actual, precision);
```

Comment:
I'd be tempted to just inline this, I don't think the separate method is helpful, it just hides what's going on. Your call obviously